src/txn/txn_ckpt.c

```diff
@@ -699,11 +684,17 @@ __txn_checkpoint_can_skip(
         return (0);
 
     /*
-     * It isn't currently safe to skip timestamp checkpoints - see WT-4958. We should fix this so we
-     * can skip timestamp checkpoints if they don't have new content.
+     * If the checkpoint is using timestamps, and the stable timestamp hasn't been updated since the
+     * last checkpoint there is nothing more that could be written. Except when a non timestamped
+     * file has been modified, as such if the connection has been modified it is currently unsafe to
+     * skip checkpoints.
      */
-    if (use_timestamp)
+    if (!conn->modified && use_timestamp && txn_global->has_stable_timestamp &&
```

Comment:
Isn't this entire check redundant? Immediately below you have:
```
    if (!conn->modified) {
        *can_skipp = true;
        return (0);
    }
```

Reply:
Thanks, fixed.